### PR TITLE
generate docstring for config, deplyer, frameworks

### PR DESCRIPTION
**Pull request by hamedhsn**
**What this PR does / why we need it**:
#326 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/fairing/354)

### CODE REVIEW

File: fairing/config.py [hunk 1]
```diff
@@ -45,10 +45,12 @@
 
 
 class Config(object):
+    """ """
```
Comment (jinchihe) [inline fairing/config.py]: Remove this? better to add some introduction for class.

File: fairing/config.py [hunk 2]
```diff
@@ -62,39 +64,65 @@ def reset(self):
         self._deployer_kwargs = {}
 
     def set_preprocessor(self, name=None, **kwargs):
+        """
+
+        :param name: preprocessor name(Default value = None)
+        :param **kwargs:
```
Comment (jinchihe) [inline fairing/config.py]: I hesitated if need to remove this, not sure what best practice for kwargs.

File: fairing/config.py [hunk 3]
```diff
@@ -62,39 +64,65 @@ def reset(self):
...
         if fn is None:
             raise Exception('Preprocessor name not found: {}\nAvailable preprocessor: {}'.format(
                 self._preprocessor_name, list(preprocessor_map.keys())))
         return fn(**self._preprocessor_kwargs)
 
     def set_builder(self, name=DEFAULT_BUILDER, **kwargs):
+        """set the builder
+
+        :param name: builder name (Default value = DEFAULT_BUILDER)
+        :param **kwargs:
```
Comment (jinchihe) [inline fairing/config.py]: Same here.

File: fairing/deployers/gcp/gcp.py [hunk 4]
```diff
@@ -6,10 +6,10 @@
 from googleapiclient import discovery
 from googleapiclient import errors
 
+
 class GCPJob(DeployerInterface):
     """Handle submitting training job to GCP.
-    Attributes:
-        project_id: Google Cloud project ID to use.
+     project_id: Google Cloud project ID to use.
```
Comment (jinchihe) [inline fairing/deployers/gcp/gcp.py]: format is not consistent here.

File: fairing/deployers/job/job.py [hunk 5]
```diff
@@ -12,13 +12,19 @@
...
 
+
 class Job(DeployerInterface): #pylint:disable=too-many-instance-attributes
     """Handle all the k8s' template building for a training
-    Attributes:
-        namespace: k8s namespace where the training's components
-            will be deployed.
-        runs: Number of training(s) to be deployed. Hyperparameter search
-            will generate multiple jobs.
+        Attributes:
```
Comment (jinchihe) [inline fairing/deployers/job/job.py]: The format is not consistent with others.

File: fairing/deployers/kfserving/kfserving.py [hunk 6]
```diff
@@ -12,22 +12,21 @@
...
-        labels: Labels for the kfservice, separate with commas if have more than one.
-        annotations: Annotations for the kfservice, separate with commas if have more than one.
-        custom_default_spec: A flexible custom default specification for arbitrary customer
-                            provided containers.
-        custom_canary_spec: A flexible custom canary specification for arbitrary customer
-                            provided containers.
-        stream_log: Show log or not when kfservice started, defaults to True.
-        cleanup: Delete the kfserving or not, defaults to False.
+    """Serves a prediction endpoint using Kubeflow KFServing.
+            Attributes:
```
Comment (jinchihe) [inline fairing/deployers/kfserving/kfserving.py]: The format is not consistent with others.

File: fairing/deployers/serving/serving.py [hunk 7]
```diff
@@ -10,12 +10,13 @@
 
 logger = logging.getLogger(__name__)
 
-class Serving(Job):
-    """
-    Serves a prediction endpoint using Kubernetes deployments and services
 
+class Serving(Job):
+    """Serves a prediction endpoint using Kubernetes deployments and services
+
```
Comment (jinchihe) [inline fairing/deployers/serving/serving.py]: The format is not consistent with others.

File: fairing/deployers/tfjob/tfjob.py [hunk 8]
```diff
@@ -6,7 +6,20 @@
 
 logger = logging.getLogger(__name__)
 
+
 class TfJob(Job):
+    """ Handle all the k8s' template building to create tensorflow training job using Kubeflow TFOperator
```
Comment (jinchihe) [inline fairing/deployers/tfjob/tfjob.py]: same here

File: fairing/frameworks/lightgbm.py [hunk 9]
```diff
@@ -86,12 +115,12 @@ def _generate_entrypoint(copy_files_before, copy_files_after, config_file,
         for rank, files in copy_patitioned_files.items():
             buf.append("\t{})".format(rank))
             buf.extend(
-                ["\t\t"+cmd for cmd in _get_commands_for_file_ransfer(files)])
+                ["\t\t" + cmd for cmd in _get_commands_for_file_transfer(files)])
```
Comment (jinchihe) [inline fairing/frameworks/lightgbm.py]: Why change here, the PR is only for update docString, for others, pls create another PR for this.

File: fairing/frameworks/lightgbm.py [hunk 10]
```diff
@@ -86,12 +115,12 @@ def _generate_entrypoint(copy_files_before, copy_files_after, config_file,
...
             buf.append("\t\t;;")
         buf.append("esac")
 
     # copying files that are common to all workers
-    buf.extend(_get_commands_for_file_ransfer(copy_files_before))
+    buf.extend(_get_commands_for_file_transfer(copy_files_before))
```
Comment (jinchihe) [inline fairing/frameworks/lightgbm.py]: Why change here, the PR is only for update docString, for others, pls create another PR for this.

**Comment by jinchihe** [comment]
pls rebase and then we can test, some presubmit problem has been fixed.

**Comment by jinchihe** [comment]
@hamedhsn Could you please get chance to update the PR? Thanks a lot!

**Comment by jinchihe** [comment]
@hamedhsn Could you please rebase to resolve conflicting and address comments? thanks.

File: fairing/config.py (hunk 2 above)
Comment (hamedhsn) [reply fairing/config.py]: lets remove it

File: fairing/deployers/gcp/gcp.py (hunk 4 above)
Comment (hamedhsn) [reply fairing/deployers/gcp/gcp.py]: done

File: fairing/frameworks/lightgbm.py (hunk 9 above)
Comment (hamedhsn) [reply fairing/frameworks/lightgbm.py]: will revert

**Comment by hamedhsn** [comment]
@jinchihe have a look should be good now.

**Comment by hamedhsn** [comment]
@jinchihe all good?

**Comment by jinchihe** [comment]
@hamedhsn So sorry for later, could you please rebase to resolve the conflicts, let merge that then, great thanks!

**Comment by hamedhsn** [comment]
@jinchihe lets merge before another conflict pls

**Comment by jinchihe** [comment]
/lgtm
/approve

@hamedhsn Thanks again!

**Pull request merged**